### PR TITLE
Maintenance/animation stuff

### PR DESCRIPTION
**Pull request by h3nx**

### CODE REVIEW

File: Sail/src/Sail/entities/EntityFactory.cpp [hunk 1]
```diff
@@ -56,21 +48,34 @@ Entity::SPtr EntityFactory::CreateMyPlayer(Netcode::PlayerID playerID, size_t li
...
-	candle->addComponent<RealTimeComponent>(); // Player candle will have its position updated each frame
-	candle->getComponent<CandleComponent>()->setOwner(playerID);
-	myPlayer->addChildEntity(candle);
+	myPlayer->getComponent<ModelComponent>()->renderToGBuffer = false;
+	myPlayer->addComponent<MovementComponent>()->constantAcceleration = glm::vec3(0.0f, -9.8f, 0.0f);
 
-	myPlayer->getComponent<TransformComponent>()->setStartTranslation(glm::vec3(1.6f, 0.9f, 1.f) + spawnLocation);
+	for (Entity::SPtr& c : myPlayer->getChildEntities()) {
+		if (c->getName() == myPlayer->getName() + "Candle") {
+			c->addComponent<CandleComponent>();
```
Comment (Skratzy) [inline Sail/src/Sail/entities/EntityFactory.cpp]: This code segment is shared in two places, could probably be extracted into a function

File: Sail/src/Sail/entities/components/AnimationComponent.h [hunk 2]
```diff
@@ -72,7 +72,17 @@ class AnimationComponent : public Component<AnimationComponent> {
 	
 	Entity* rightHandEntity;
 	Entity* leftHandEntity;
-
+	void setAnimation(const unsigned int index) {
+		animationIndex = index;
+		if (animationIndex > m_stack->getAnimationCount()) {
```
Comment (Skratzy) [inline Sail/src/Sail/entities/components/AnimationComponent.h]: Should probably log something cause we're supposed to know how many animations there are?

File: Sail/src/Sail/entities/systems/input/GameInputSystem.cpp [hunk 3]
```diff
@@ -94,12 +94,14 @@ void GameInputSystem::processKeyboardInput(const float& dt) {
...
 			// Else do normal movement
-		} else {
+		} 
+		else {
 			auto collision = e->getComponent<CollisionComponent>();
 			auto movement = e->getComponent<MovementComponent>();
 			auto speedLimit = e->getComponent<SpeedLimitComponent>();
 			auto audioComp = e->getComponent<AudioComponent>();
 
+			//movement->forwardVel = glm::vec3(playerMovement.rightMovement, playerMovement.upMovement, playerMovement.forwardMovement);
```
Comment (Skratzy) [inline Sail/src/Sail/entities/systems/input/GameInputSystem.cpp]: Perhaps remove this code if it's not needed

File: Sail/src/Sail/entities/components/AnimationComponent.h (hunk 2 above)
Comment (h3nx) [reply Sail/src/Sail/entities/components/AnimationComponent.h]: nonissue, there is an enum in animationChanger which is used.

**Pull request merged**